File: 프로그래머스/2/87946. 피로도/피로도.java
import java.util.*;

class Solution {
    
    int answer = -1;
    
    public void backtracking(int[][] dungeons, List<Integer> cur, int left) {
        answer = Math.max(answer, cur.size());
        for(int i=0;i<dungeons.length;i++) {
            int need = dungeons[i][0];
            int use = dungeons[i][1];
            if(need <= left && !cur.contains(i)) {
                cur.add(i);
                backtracking(dungeons,cur,left-use);
                cur.remove(cur.size()-1);
            }
        }
    }
    
    public int solution(int k, int[][] dungeons) {
       backtracking(dungeons, new ArrayList<>(), k);
       return answer;
    }
}

File: 프로그래머스/2/87946. 피로도/피로도.swift
import Foundation

func solution(_ k:Int, _ dungeons:[[Int]]) -> Int {
    
    var answer = 0
    
    func backtracking(_ cur:[Int], _ now:Int) {
        answer = max(answer, cur.count)
        for (i, dungeon) in dungeons.enumerated() {
            let need = dungeons[i][0]
            if(!cur.contains(i) && need <= now) {
                var nxt = cur
                nxt.append(i)
                backtracking(nxt,now-dungeons[i][1])
            }
        }
    }
    
    backtracking([], k)
    
    return answer
}